source/docs/network_filter_fuzzing.md

```diff
@@ -0,0 +1,119 @@
+Each test case is a file under the folder [network_readfilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_readfilter_corpus) or [network_writefilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_writefilter_corpus). It consists of two parts: `config` and `actions`. 
+`config` is the protobuf to instantiate a filter, and `actions` are sequences of actions to take in order to test the filter. 
+An example for testing MongoProxy filter:
+```
+config {
+  name: "envoy.filters.network.mongo_proxy"
+  typed_config {
+    type_url: "type.googleapis.com/envoy.extensions.filters.network.mongo_proxy.v3.MongoProxy"
+    value: "\n\001\\\032\t\032\002\020\010\"\003\010\200t \001"
+  }
+}
+actions {
+  on_new_connection {
+  }
+}
+actions {
+  on_data {
+    data: "\120\0\0\0\1\0\0\0\1\0\0\0\324\7\0\0\4\0\0\0\164\145\163\164\56\164\145\163\164\0\24\0\0\0\377\377\377\377\52\0\0\0\2\163\164\162\151\156\147\137\156\145\145\144\137\145\163\143\0\20\0\0\0\173\42\146\157\157\42\72\40\42\142\141\162\12\42\175\0\0"
+  }
+}
+actions {
+  advance_time {
+    milliseconds: 10000
+  }
+}
+```
+* `config.name` is the name of the filter. 
+* `config.typed_config.type_url` is the type url of the filter config API. 
+* `config.typed_config.value` is the serialized string of the config protobuf, and in C++ we can call`config.SerializeAsString()` to obtain this. This string may contain special characters. Recommend using octal or hexadecimal sequence for the string.
+* `actions.on_data.data` (or `actions.on_write.data`) is the buffer parameter `data`(in string format) for testing ReadFilter's method onData() (or for testing WriteFilter's method onWrite()). This string may contain special characters. Recommend using octal or hexadecimal sequence for the string.
+* `actions.on_data.end_stream` (or `actions.on_write.end_stream`) is the bool parameter `end_stream` for testing ReadFilter's method onData() (or for testing WriteFilter's method onWrite()).
+* `actions.on_new_connection` is an action to call `onNewConnection` method of a ReadFilter.
+* `actions.advance_time.milliseconds` is the duration in milliseconds for the simulatedSystemTime to advance by.
+For more details, see the APIs for [ReadFilter Fuzz Testcase](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_readfilter_fuzz.proto) and  [WriteFilter Fuzz Testcase](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_writefilter_fuzz.proto).
+
+## Convert a unit test case to a fuzz test case manually (experimental)
+Unit test cases usually leads the filter into interesting states. Currently there is no automatic way to convert a unit test case into a fuzz test case. However, there is a way to convert it manually.
```

Comment:
You do say it is experimental, but you would be surprised how many people will miss that.  Say something like:

This section explains an advanced technique to generate a corpus from unit tests.  It is an optional step for users who want to generate the highest possible coverage.

Reply:
> You do say it is experimental, but you would be surprised how many people will miss that. Say something like:
> 
> This section explains an advanced technique to generate a corpus from unit tests. It is an optional step for users who want to generate the highest possible coverage.

I see, thanks!